arch/loongarch/kernel/irq.c

```diff
@@ -135,3 +136,19 @@ void __init init_IRQ(void)
 
 	set_csr_ecfg(ECFGF_IP0 | ECFGF_IP1 | ECFGF_IP2 | ECFGF_IPI | ECFGF_PMC);
 }
+
+DEFINE_PER_CPU(int, irq_nesting);
+
+#ifdef CONFIG_IRQ_PIPELINE
+asmlinkage int 
+handle_arch_irq_pipelined(struct pt_regs *regs)
```

Comment:
Here needs a `#else` branch for `handle_arch_irq_pipelined`. It is better to follow the original code pattern.